All_years/04-Social_Care/04-SDS_data.R

```diff
@@ -0,0 +1,167 @@
+  # Pivot longer on sds option variables
+  pivot_longer(
+    cols = contains("sds_option_"),
+    names_to = "sds_option",
+    names_prefix = "sds_option_",
+    names_transform = list(sds_option = ~ paste0("SDS-", .x)),
+    values_to = "received",
+    values_transform = list(received = as.integer)
+  ) %>%
+  # Only keep rows where they received a package and remove duplicates
+  filter(received == 1) %>%
+  distinct() %>%
+  # Include source variables
+  mutate(
+    recid = "SDS",
+    mid_fy = as.Date(paste0(period, "-09-30")),
+    # create age variable
+    age = floor(time_length(interval(dob, mid_fy), "years")),
+    # Create person id variable
+    person_id = glue::glue("{sending_location}-{social_care_id}"),
+    # Use function for creating sc send lca variables
+    sc_send_lca = convert_sc_sl_to_lca(sending_location)
+  ) %>%
+  # when multiple social_care_id from sending_location for single CHI
+  # replace social_care_id with latest
+  group_by(sending_location, chi) %>%
+  mutate(latest_sc_id = last(social_care_id)) %>%
+  # count changed social_care_id
+  mutate(
+    changed_sc_id = !is.na(chi) & social_care_id != latest_sc_id,
+    social_care_id = if_else(changed_sc_id, latest_sc_id, social_care_id)
+  ) %>%
+  ungroup()
+
+
+merge_eps <- sds_full_clean %>%
+  # Use lazy_dt() for faster running of code
+  dtplyr::lazy_dt() %>%
+  group_by(sending_location, social_care_id, sds_option) %>%
+  arrange(period, record_keydate1, .by_group = TRUE) %>%
+  # Create a flag for episodes that are going to be merged
+  # Create an episode counter
+  mutate(
+    flag = (record_keydate1 > lag(record_keydate2)) %>%
+      replace_na(TRUE),
+    episode_counter = cumsum(flag)
+  ) %>%
+  # Group by episode counter and merge episodes
+  group_by(episode_counter, .add = TRUE) %>%
+  summarise(
+    sc_latest_submission = last(period),
+    record_keydate1 = min(record_keydate1),
+    record_keydate2 = max(record_keydate2),
+    sending_location = last(sending_location),
+    social_care_id = last(social_care_id),
+    chi = last(chi),
+    gender = last(gender),
+    dob = last(dob),
+    postcode = last(postcode),
+    sds_option = last(sds_option),
+    recid = last(recid),
+    age = last(age),
+    person_id = last(person_id),
+    sc_send_lca = last(sc_send_lca),
+    episode_counter = last(episode_counter)
+  ) %>%
+  # end of lazy_dt()
+  as_tibble()
+
+# work out sds_option_4
+sds_option_4 <- merge_eps %>%
+  group_by(sending_location, social_care_id, period) %>%
+  summarise(n_packages = n_distinct(sds_option))
+
+# Match back onto data
+outfile <- merge_eps %>%
+  left_join(sds_option_4, by = c("sending_location", "social_care_id", "period")) %>%
+  mutate(
+    sds_option_4 = if_else(n_packages >= 2, 1, 0)
+  )
+
+
+# Save outfile------------------------------------------------
+
+outfile %>%
+  # save rds file
+  readr::write_rds(path(social_care_dir, str_glue("all_sds_episodes_{latest_update}.rds")),
+    compress = "gz"
```

Comment:
```suggestion
    compress = "xz"
```
Better (and faster) compression.